packages/libs/error-stack/tests/snapshots/test_serialize__full__attachment.snap

```diff
@@ -0,0 +1,43 @@
+---
+source: tests/test_serialize.rs
+expression: report
+---
+[
+  {
+    "context": "root error",
+    "attachments": [
+      {
+        "file": "tests/common/mod.rs",
+        "line": 4,
+        "column": 5
+      },
```

Comment:
By looking at the structure it's clearly a location, but do we somehow want to tag that? Or is that too opinionated? cc @Alfred-Mountfield

Reply:
If we're expecting quite a few similar attachment things, I think it'd be nice to have a `kind` field. If we're not expecting that, then I think this is fine

Reply:
We could do:

```json
{
	"kind": "...",
	"value": {...}
}
```

The main problem is the `"kind"` value. We could use `TypeId` (but that value isn't stable across recompilations and is only valid for `'static` types), and another one would be https://doc.rust-lang.org/std/any/fn.type_name.html.

Reply:
I thought we're not deriving `Serialize`? So we could just choose the value of `"kind"` and call it `Location` or something, no?

Reply:
Again I'm not sure this is worth it unless it's a pattern we're establishing for things

Reply:
Yes, but ideally - if this were a pattern - we would automatically assign a `kind` to every attachment